cli/src/commands/split.rs

```diff
@@ -259,8 +378,20 @@ pub(crate) fn cmd_split(
     }
 
     if let Some(mut formatter) = ui.status_formatter() {
-        if num_rebased > 0 {
-            writeln!(formatter, "Rebased {num_rebased} descendant commits")?;
+        let mut num_rebased_descendants = rewritten_commits.len() + rebased_commits.len();
+        // don't count the commit generated by the split in the rebased commits
+        num_rebased_descendants -= num_new_rebased;
+        for (_, ref rewritten) in rewritten_commits {
+            if rebased_commits.contains_key(rewritten) {
+                // only count once a commit that may have been rewritten twice in the process
+                num_rebased_descendants -= 1;
```

Comment:
nit: rebasing twice would leave intermediate state in evolution history. Maybe this can be addressed later.

Reply:
I tried to call `move_commit` and then rewrite the descendant commit that may not have been touched by `move_commit`, but I wasn't always getting a consistent result.
That may be easier to do now that we have a set of tests to make sure it works properly.
If you don't mind, I'll try that again in another PR

Reply:
Yeah, in another PR. Feel free to add some TODO comment if that makes sense.

I think this can be addressed by removing predecessors records from the commit object, but we don't agree on that yet.

Reply:
done in #6661!